docs/src/tutorial/arbitrary-variables/src/rating.rs

```diff
@@ -3,40 +3,42 @@
 
 // ANCHOR: rating_struct
 #[derive(Copy, Clone)]
-pub struct Rating {
-    value: u8,
+enum Rating {
+    One,
+    Two,
+    Three,
 }
 
 impl Rating {
-    pub fn from(value: u8) -> Option<Rating> {
-        if value <= 5 { Some(Rating { value }) } else { None }
-    }
-
-    pub fn get(&self) -> u8 {
-        self.value
+    fn as_int(&self) -> u8 {
+        match self {
+            Rating::One => 1,
+            Rating::Two => 2,
+            Rating::Three => 3,
+        }
     }
 }
-
 // ANCHOR_END: rating_struct
 
 #[cfg(kani)]
 mod verification {
     use super::*;
 
     // ANCHOR: rating_invariant
-    unsafe impl kani::Invariant for Rating {
-        fn is_valid(&self) -> bool {
-            self.value <= 5
+    fn any_rating() -> Rating {
+        match kani::any() {
+            0 => Rating::One,
+            1 => Rating::Two,
+            _ => Rating::Three,
```

Comment:
This is a bit counter intuitive since it doesn't match `as_int()`.

Reply:
Hmm, I agree, but maybe that's a feature: emphasize that there's not reason for it to match.

Reply:
I personally prefer minimizing the things we have to explain in a tutorial but not a blocker.